docs/source/tutorials/SAR/SAR tutorial.ipynb

```diff
@@ -0,0 +1,1180 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "d2fff195",
+   "metadata": {
+    "scrolled": false
+   },
+   "outputs": [],
+   "source": [
+    "test_env('myoLegStairTerrainWalk-v0', episodes=1, timesteps=25)\n",
+    "show_video('test_env_temp.mp4')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "7cbc2a7d",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def get_video(env_name, model_path, env_path, episodes, video_name, determ=False, \n",
+    "            pca=None, ica=None, scaler=None, phi=None, is_sar=False, syn_nosyn=False):\n",
+    "    frames = []\n",
+    "    if is_sar:\n",
+    "        if syn_nosyn:\n",
+    "            env = SynNoSynWrapper(gym.make(env_name), ica, pca, scaler, phi)\n",
+    "        else:\n",
+    "            env = SynergyWrapper(gym.make(env_name), ica, pca, scaler)\n",
+    "    else:\n",
+    "        env = gym.make(env_name)\n",
+    "\n",
+    "\n",
+    "    camera = 'side_view'\n",
+    "        \n",
+    "    for i,__ in tqdm(enumerate(range(episodes))):              \n",
+    "        env.reset()\n",
+    "\n",
+    "        model = SAC.load(model_path)\n",
+    "        vec = VecNormalize.load(env_path, DummyVecEnv([lambda: env]))\n",
+    "        \n",
+    "        rs = 0\n",
+    "        is_solved = []\n",
+    "        done = False\n",
+    "        for _ in range(1000):\n",
+    "            o = vec.normalize_obs(env.get_obs())\n",
+    "            a, __ = model.predict(o, deterministic=determ)\n",
+    "            \n",
+    "            frame = env.sim.renderer.render_offscreen(width=640, height=480,camera_id=camera)\n",
+    "            frames.append(frame)\n",
+    "\n",
+    "            next_o, r, done, info = env.step(a)\n",
+    "            is_solved.append(info['solved'])\n",
+    "            \n",
+    "            rs+=r\n",
+    "    env.close()\n",
+    "    skvideo.io.vwrite(f'{video_name}.mp4', np.asarray(frames),outputdict={\"-pix_fmt\": \"yuv420p\"})\n",
+    "    \n",
+    "ica,pca,norm = load_locomotion_SAR()\n",
+    "env_path = 'multihill_terrain_syn_env_myoLegWalkHillyTerrain-v0_3'\n",
+    "model_path = 'multihill_terrain_syn_model_myoLegWalkHillyTerrain-v0_3.zip'\n",
+    "get_video('Test-v0',model_path,env_path,1,'test',pca=pca,ica=ica,scaler=norm,is_sar=True)\n",
+    "show_video('test.mp4')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "fc299ac0",
+   "metadata": {},
+   "source": [
+    "# SAR core functions"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c8433315",
+   "metadata": {},
+   "source": [
+    "<div style=\"display: flex; align-items: flex-start;\">\n",
+    "    <div style=\"flex: 1;\">\n",
+    "         Next, we'll define and implement the core functions for SAR. The framework follows the basic structure displayed on the right. Accordingly, we will define functions that do all of the following:\n",
```

Comment:
The flow can be improved.

From the current flow, it seems like we are going to outline the core functionality and then give the SAR algorithm. What I'll recommend is to think of it like a lecture. Give a 2-3 line overview of SAR. Follow it with a step by step implementation.

Reply:
Updated to step by step implementation that should be very intuitive to follow.

Flow:

- SAR is composed of 4 steps (explained and visualized in new figure)
- Each step, with associated functions, is labeled, introduced, explained
- After implementing each step, we turn to using it in two end-to-end examples